### PR TITLE
Checks GitHub API if homepage 404s during `brew audit --online`

### LINKED ISSUE #13789: `brew audit --online` not adding the HOMEBREW_GITHUB_API_TOKEN as HTTP Authorization Header
### `brew config` output

```shell
▶ brew config          
HOMEBREW_VERSION: 3.5.10-89-g451bded
ORIGIN: https://github.com/Homebrew/brew
HEAD: 451bded34d315f21ff8fcdd0a61c6ba052625b30
Last commit: 7 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 001186aea9d67bb1829448098ec89f0b363d65f4
Core tap last commit: 6 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_MAKE_JOBS: 8
HOMEBREW_NO_AUTO_UPDATE: set
https_proxy: 127.0.0.1:8081
Homebrew Ruby: 2.6.8 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8_1/bin/ruby
CPU: octa-core 64-bit icelake
Clang: 13.0.0 build 1300
Git: 2.30.1 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.6.8-x86_64
CLT: 12.5.0.22.9
Xcode: 13.0
```

### Verification

- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.

### What were you trying to do (and why)?

I bumped a private Formula.  All worked.

I set the HOMEBREW_GITHUB_API_TOKEN as a local environment variable.

OK:

`brew upgrade foo-private-cli`

OK:

`brew audit foo-private-cli --git --skip-style -d`

NOK:

`brew audit foo-private-cli --online --git --skip-style -d`

### What happened (include all command output)?

```
  * The homepage URL https://github.com/foo/foo-private-cli is not reachable (HTTP status code 404)
  * Stable: The source URL https://github.com/foo/foo-private-cli/releases/download/0.1.1/foo-private-cli is not reachable (HTTP status code 404)
Error: 2 problems in 1 formula detected
```

### What did you expect to happen?

As I can `brew install`, `upgrade` and `reinstall` from these private repos, I know my `HOMEBREW_GITHUB_API_TOKEN` works and there are no "Single Sign On" (SSO issues with the GitHub Personal Access Token.

image

### PR DESCRIPTION
**Pull request by troymccabe**
- [X] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [X] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [X] Have you successfully run `brew style` with your changes locally?
- [X] Have you successfully run `brew typecheck` with your changes locally?
- [X] Have you successfully run `brew tests` with your changes locally?

-----

Fixes #13789

### CODE REVIEW

File: Library/Homebrew/utils/curl.rb [hunk 1]
```diff
@@ -290,7 +290,25 @@ def curl_check_http_content(url, url_type, specs: {}, user_agents: [:default],
           url_protected_by_cloudflare?(response) || url_protected_by_incapsula?(response)
         end
 
-        return "The #{url_type} #{url} is not reachable (HTTP status code #{details[:status_code]})"
+        # https://github.com/Homebrew/brew/issues/13789
+        # If the `:homepage` of a formula is private, it will fail an `audit`
+        # since there's no way to specify a `strategy` with `using:` and
+        # Github does not authorize access to the web ui using token
```
Comment (MikeMcQuaid) [inline Library/Homebrew/utils/curl.rb:296]:
```suggestion
        # GiHhub does not authorize access to the web UI using token
```

File: Library/Homebrew/utils/curl.rb [hunk 2]
```diff
@@ -290,7 +290,25 @@ def curl_check_http_content(url, url_type, specs: {}, user_agents: [:default],
...
+        #
+        # Strategy:
+        # If the `:homepage` 404s, it's a github link, and we have a token--
```
Comment (MikeMcQuaid) [inline Library/Homebrew/utils/curl.rb:299]:
```suggestion
        # If the `:homepage` 404s, it's a GitHub link, and we have a token then
```

File: Library/Homebrew/utils/curl.rb [hunk 3]
```diff
@@ -290,7 +290,25 @@ def curl_check_http_content(url, url_type, specs: {}, user_agents: [:default],
...
+        # check the API for the repository existing (which does use tokens)
```
Comment (MikeMcQuaid) [inline Library/Homebrew/utils/curl.rb:300]:
```suggestion
        # check the API for the repository (which does use tokens)
```

**Review by MikeMcQuaid — COMMENTED** [review]
Looks good so far! Nice work.

File: Library/Homebrew/utils/curl.rb (hunk 3 above)
Comment (troymccabe) [reply Library/Homebrew/utils/curl.rb]: Also moved parenthetical next to "API" to be more clear that it's the API that uses tokens, not the repository. Let me know if that's alright

**Comment by troymccabe** [comment]
🙏 Thanks again for the review! Please lmk if there's anything else that needs to change

**Comment by troymccabe** [comment]
Of course! I'm poking through the other `help_wanted` issues to see if there's others I can pick up. Has been fun!

Anything I should change on my end for the `codecov/project` check? Looks like a bunch of unrelated coverage % change in codecov: https://github.com/Homebrew/brew/pull/13907/checks?check_run_id=8472344053

**Pull request merged**